network-api/networkapi/wagtailpages/tests/buyersguide/test_article_page.py

```diff
@@ -29,6 +31,8 @@ def setUpTestData(cls):
         cls.content_index = buyersguide_factories.BuyersGuideEditorialContentIndexPageFactory(
             parent=cls.buyersguide_homepage,
         )
+        edit_handler = pagemodels.BuyersGuideArticlePage.get_edit_handler()
+        cls.article_page_form = edit_handler.get_form_class()
```

Comment:
Can you add some context why we need to go through this `get_edit_handler` method instead of using the custom form class directly? Where did you find this method? It's not documented on the model reference: https://docs.wagtail.org/en/v2.16.3/reference/pages/model_reference.html

Reply:
https://github.com/wagtail/wagtail/blob/5fc191b116ee835762c282e13a5829fa873cc516/wagtail/admin/tests/test_edit_handlers.py#L145-L156

Reply:
Hi Tibor! 

I found this method in the wagtail source code's test suite, in a test called [test_get_form_for_page_with_custom_base](https://github.com/wagtail/wagtail/blob/main/wagtail/admin/tests/test_edit_handlers.py#L262-L272)

```python
    def test_get_form_for_page_with_custom_base(self):
        """
        ValidatedPage sets a custom base_form_class. This should be used as the
        base class when constructing a form for ValidatedPages
        """
        edit_handler = ValidatedPage.get_edit_handler()
        GeneratedValidatedPageForm = edit_handler.get_form_class()

        # The generated form should inherit from ValidatedPageForm, because
        # ValidatedPage.base_form_class == ValidatedPageForm
        self.assertTrue(issubclass(GeneratedValidatedPageForm, ValidatedPageForm))
```

Reply:
Also have seen something similar when Rich was testing the validity of forms for [BuyersGuide Products](https://github.com/mozilla/foundation.mozilla.org/blob/137337624947b0f1fc71d612e78eaed92195ab49/network-api/networkapi/wagtailpages/tests.py#L471-L484)

```python
class BuyersGuideProductCategoryTest(TestCase):
    def setUp(self):
        edit_handler = get_snippet_edit_handler(BuyersGuideProductCategory)
        self.form_class = edit_handler.get_form_class()

    def test_cannot_have_duplicate_name(self):
        BuyersGuideProductCategory.objects.create(name="Cat 1")

        form = self.form_class(
            data=self.generate_form_data({"name": "Cat 1", "sort_order": 1}),
        )

        self.assertFalse(form.is_valid())
        self.assertEqual(1, len(form.errors))
        self.assertIn("name", form.errors)
```

Reply:
I think it's important to know why are using certain features because it can lead to unexpected outcomes.

Looking at this for another moment, I think it is important to the `get_edit_handler` because our form is not fully featured. It does not define a model or any fields. It would therefore not be usable on it's own.

Reply:
Before we just more on, there is still the question of what happens in the "valid" test without your changes / the base class. Do we already see that behavior that the form is "never valid"?